src/Core/Utilities/StrictEmailAddressAttribute.cs

```diff
@@ -18,7 +18,7 @@ public override bool IsValid(object value)
                 return false;
             }
 
-            var illegalChars = @"[\s<>()]";
+            var illegalChars = @"[\s<>();]|[.,]$";
```

Comment:
It seems like this is going to end up being whack-a-mole. 

Does is make sense just using something like this:
```c#
System.Net.Mail.MailAddress(emailAddress).Address == emailAddress
```
?

Reply:
This still doesn't catch the trailing `.`, btw

Reply:
That would work great for MailKit, which exposes its own `MailboxAddress.Parse` method - so we could call that and compare the result as you suggest.

However, Amazon SES doesn't seem to expose its own parsing method (at least that I could find), so we couldn't use it there. We'd have to use MailKit or `System.Net.Mail` as analogues and hope that they have similar parsing logic - which they probably do, but then it does feel like we're crossing our fingers. In this case, I think you'd still want the regex there for some certainty (you'd at least need it to capture the trailing `.` if nothing else) - and then that feels like we're doubling up on checks.

What do you think?

Reply:
I would presume it would be better to apply a general validator from MailKit, and then ALSO validate any illegal characters also rejected upon by SES, using the most restrictive measure.

See: https://stackoverflow.com/a/1903368/14067465

See Also (for SES errors regarding email address): https://docs.aws.amazon.com/ses/latest/DeveloperGuide/troubleshoot-error-messages.html

>**There are non-ASCII characters in the email address** — The email address string must be 7-bit ASCII. If you want to send to or from email addresses that contain Unicode characters in the domain part of an address, you must encode the domain using Punycode. Punycode is not permitted in the local part of the email address (the part before the @ sign) nor in the "friendly from" name. If you want to use Unicode characters in the "friendly from" name, you must encode the "friendly from" name using MIME encoded-word syntax, as described in [Sending raw email using the Amazon SES API](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-raw.html). For more information about Punycode, see [RFC 3492](http://tools.ietf.org/html/rfc3492).